qcodes/tests/dataset/dond/test_doNd.py

```diff
@@ -1863,3 +910,641 @@ def test_dond_2d_multi_datasets_with_callable_output_data(
+    )
+    assert len(datasets[1].description.shapes) == 1
+    assert datasets[1].description.shapes[
+        "dummyinstrument_ChanB_dummy_parameter_with_setpoints"
+    ] == (outer_shape, inner_shape, n_points_b)
+
+
+def test_dond_sweeper_combinations(_param_set, _param_set_2, _param):
+    outer_shape = 10
+    inner_shape = 15
+
+    a = ManualParameter("a", initial_value=0)
+    b = ManualParameter("b", initial_value=0)
+    c = ManualParameter("c", initial_value=0)
+    d = ManualParameter("d", initial_value=0)
+    sweep_a = LinSweep(a, 0, 3, outer_shape)
+    sweep_b = LinSweep(b, 5, 7, outer_shape)
+    sweep_c = LinSweep(c, 8, 12, outer_shape)
+    sweep_d = LinSweep(d, 13, 16, inner_shape)
+
+    multi_sweep = TogetherSweep(sweep_a, sweep_b, sweep_c)
+
+    sweeper = _Sweeper([multi_sweep, sweep_d], [])
+
+    sweep_groups = sweeper.sweep_groupes
+
+    expected_sweeper_groups = (
+        (a, d),
+        (b, d),
+        (c, d),
+        (a, b, d),
+        (a, c, d),
+        (b, c, d),
+        (a, b, c, d),
+    )
+
+    assert len(expected_sweeper_groups) == len(sweep_groups)
+    for g in expected_sweeper_groups:
+        assert g in sweep_groups
+
+
+def test_sweep_int_vs_float():
+
+    float_param = ManualParameter("float_param", initial_value=0.0)
+    int_param = ManualParameter("int_param", vals=Ints(0, 100))
+
+    dataset, _, _ = dond(ArraySweep(int_param, [1, 2, 3]), float_param)
+    assert dataset.parameters == "int_param,float_param"
+    assert dataset.cache.data()["float_param"]["int_param"].dtype.kind == "i"
+
+
+def test_error_no_measured_parameters():
+    float_param = ManualParameter("float_param", initial_value=0.0)
+    int_param = ManualParameter("int_param", vals=Ints(0, 100))
+
+    with pytest.raises(ValueError, match="No parameters to measure supplied"):
+        dond(ArraySweep(int_param, [1, 2, 3]), ArraySweep(float_param, [1.0, 2.0, 3.0]))
+
+
+def test_error_measured_grouped_and_not_grouped():
+    param_1 = ManualParameter("param_1", initial_value=0.0)
+    param_2 = ManualParameter("param_2", initial_value=0.0)
+    param_3 = ManualParameter("param_3", initial_value=0.0)
+
+    with pytest.raises(
+        ValueError, match="Got both grouped and non grouped parameters to measure in"
+    ):
+        dond(LinSweep(param_1, 0, 10, 10), param_2, [param_3])
+
+
+def test_post_action(mocker):
+    param_1 = ManualParameter("param_1", initial_value=0.0)
+    param_2 = ManualParameter("param_2", initial_value=0.0)
+
+    post_actions = (mocker.MagicMock(),)
+    dond(LinSweep(param_1, 0, 10, 10, post_actions=post_actions), param_2)
+
+    post_actions[0].assert_called_with()
+
+
+def test_extra_log_info(caplog):
+
+    param_1 = ManualParameter("param_1", initial_value=0.0)
+    param_2 = ManualParameter("param_2", initial_value=0.0)
+
+    log_message = "FOOBAR"
+    with caplog.at_level(level=logging.INFO):
+        dond(LinSweep(param_1, 0, 10, 10), param_2, log_info=log_message)
```

Comment:
perhaps add a test where NOT passing log_info results in the "using dond" in the logs?